### PR TITLE
Added the company model and some associations with the user model

### PR DESCRIPTION
**Pull request by nangga-at**
1. the company model consists of a field name of type string and a field description of type text.
2. the company model is associated with the user model with "has_and_belongs_to_many" association. schema updated accordingly.
3. tests are not done yet.

### CODE REVIEW

File: server/app/models/user.rb [hunk 1]
```diff
@@ -16,6 +16,8 @@ def as_json(options={})
       super(options)
     end
 
+  has_and_belongs_to_many :companies
```
Comment (ryanzesch) [inline server/app/models/user.rb:19]: I think we want a user to 'belong to' a company - I don't think we want a user to be able to have multiple companies, right?

File: server/db/migrate/20221020053651_create_join_table_companies_users.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateJoinTableCompaniesUsers < ActiveRecord::Migration[7.0]
+  def change
+    create_join_table :companies, :users do |t|
+      # t.index [:company_id, :user_id]
+      # t.index [:user_id, :company_id]
+      # t.index :company_id
```
Comment (ryanzesch) [inline server/db/migrate/20221020053651_create_join_table_companies_users.rb:6]: I can't quite tell, but I think you modified the schema.rb when you meant to be changing this file to change the fields of the companies model

**Review by insoochung — CHANGES_REQUESTED** [review]
You will need routes such as:
- Getting all company entries (e.g. GET /company/)
- Getting a company by ID (e.g. GET /company/:id)
- Adding/Editing/Deleting a company
- Adding a user to a company
- Removing a user from a company

Depending on what attributes the company model has you might need a bit more. And of course you should add methods that support these routes. I recommend that you 1) first implement this methods and test them with rspec, and 2) add routes and test them with cucumber.

File: server/db/schema.rb [hunk 3]
```diff
@@ -10,15 +10,22 @@
...
+
   create_table "users", force: :cascade do |t|
     t.string "email"
     t.string "password_digest"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.string "usertype"
     t.boolean "email_confirmed", default: false
     t.string "confirm_token"
-    t.string "usertype", default: "student"
```
Comment (insoochung) [inline server/db/schema.rb:21]: One quickie: could we have the default value back?

Comment (insoochung) [reply server/db/schema.rb:21]: My current unmerged PR kinda depends on it.

**Review by insoochung — DISMISSED** [review]

File: server/app/controllers/users_controller.rb [hunk 4]
```diff
@@ -248,6 +248,21 @@ def delete_from_meeting
...
+    # should i use foreign key instead of :id?
+    @user = User.find_by_id(params[:id])
+    @company = Company.find_by_id(params[:id])
+    @company.users << @user
+  end
+
+  def delete_from_company
+    @user = User.find_by_id(params[:id])
+    @company = Company.find_by_id(params[:id])
+    @company.users.delete(@user)
```
Comment (ryanzesch) [inline server/app/controllers/users_controller.rb:263]: Please double check that this doesn't delete the user itself

**Review by ryanzesch — APPROVED** [review]
Looks good. Next time, add some rspec tests :)

File: server/db/schema.rb (hunk 3 above)
Comment (ryanzesch) [reply server/db/schema.rb]: solved

**Pull request merged**